### PR TITLE
dotCMS/core#16452 Create dot-date-time

### PR DESCRIPTION
**Pull request by hmoreras**

### CODE REVIEW

File: projects/dotcms-field-elements/src/components/dot-date-time/dot-date-time.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+                    label="Date and Time:"
+                    name="dt01"
+                    value="2019-01-20 11:10:00"
+                    hint="date time hint"
+                    required
+                    required-message="Required Date Time"
+                    validation-message="Value out of range"
+                    min="2000-01-01 01:00:00"
+                    max="2020-10-30 23:30:59"
+                    step="2 2"
```
Comment (alfredo-dotcms) [inline projects/dotcms-field-elements/src/components/dot-date-time/dot-date-time.e2e.ts]: what's this? `2 2`

File: projects/dotcms-field-elements/src/components/dot-date-time/dot-date-time.tsx [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    @Prop() required: boolean;
+    @Prop() requiredMessage: string;
+    @Prop() validationMessage: string;
+    @Prop() disabled = false;
+    /** format should be year-month-day hour:minute:second | year-month-day | hour:minute:second */
+    @Prop() min: string;
+    /** format should be year-month-day hour:minute:second | year-month-day | hour:minute:second */
+    @Prop() max: string;
+    /** two values separates by a space, first goes to date, second to time */
+    @Prop() step: string;
```
Comment (alfredo-dotcms) [inline projects/dotcms-field-elements/src/components/dot-date-time/dot-date-time.tsx]: why separated by `space`, I think, were splitting things with comma

File: projects/dotcms-field-elements/src/components/dot-date-time/dot-date-time.e2e.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+                    dotTouched: true,
+                    dotValid: true
+                }
+            });
+            expect(spyValueChange).toHaveReceivedEventDetail({
+                name: 'dt01',
+                value: '2019-02-20 05:10:00'
+            });
+            expect(evt_statusChange.length).toEqual(0);
+            expect(evt_valueChange.length).toEqual(0);
```
Comment (alfredo-dotcms) [inline projects/dotcms-field-elements/src/components/dot-date-time/dot-date-time.e2e.ts]: please, add this test here:
https://github.com/dotCMS/core-web/blob/084f1acfd2625cafdb7dc9b2699d7ffd115abbc4/projects/dotcms-field-elements/src/components/dot-key-value/dot-key-value.tsx#L64

File: projects/dotcms-field-elements/src/components/dot-date-time/dot-date-time.e2e.ts (hunk 1 above)
Comment (fmontes) [reply projects/dotcms-field-elements/src/components/dot-date-time/dot-date-time.e2e.ts]: Maybe is better `2,2`?

File: projects/dotcms-field-elements/src/components/dot-date-time/dot-date-time.e2e.ts [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        expect(element.outerHTML).toBe(tagsRenderExpected);
+    });
+
+    it('should be valid, touched & dirty on date change', async () => {
+        await dateInput.press('2');
+        await page.waitForChanges();
+        expect(element.classList.contains('dot-valid')).toBe(true);
```
Comment (fmontes) [inline projects/dotcms-field-elements/src/components/dot-date-time/dot-date-time.e2e.ts]: Can we check all this 3 in 1 expect?

File: projects/dotcms-field-elements/src/components/dot-date-time/dot-date-time.tsx (hunk 2 above)
Comment (fmontes) [reply projects/dotcms-field-elements/src/components/dot-date-time/dot-date-time.tsx]: give-that-man-2-beer-com-14299358

File: projects/dotcms-field-elements/src/components/dot-input-calendar/dot-input-calendar.tsx [hunk 5]
```diff
@@ -135,7 +134,11 @@ export class DotInputCalendarComponent {
     private emitValueChange(): void {
         this._valueChange.emit({
             name: this.name,
-            value: this.value
+            value: this.formattedValue()
         });
     }
+
+    private formattedValue(): string {
+        return this.value.length === 5 ? `${this.value}:00` : this.value;
```
Comment (fmontes) [inline projects/dotcms-field-elements/src/components/dot-input-calendar/dot-input-calendar.tsx]: Are we sure this is safe enough?

Comment (fmontes) [reply projects/dotcms-field-elements/src/components/dot-input-calendar/dot-input-calendar.tsx]: What is 5?

Comment (hmoreras) [reply projects/dotcms-field-elements/src/components/dot-input-calendar/dot-input-calendar.tsx]: possible values are `12:34:32` | `12:34` | `2000-02-02`. The time input remove the seconds when they are zero. is to force `HH:mm:ss` format

File: projects/dotcms-field-elements/src/components/dot-date-time/dot-date-time.e2e.ts (hunk 3 above)
Comment (hmoreras) [reply projects/dotcms-field-elements/src/components/dot-date-time/dot-date-time.e2e.ts]: done

**Pull request merged**